app/locales/languages.js

```diff
@@ -68,6 +72,41 @@ export function findUserLang(callback) {
   });
 }
 
+// This function is for checking/change current app direction
+export function languageDirectionHandler(language, navigation) {
+  // check if is not rtl and user choose rtl language
+  if (!I18nManager.isRTL && rtlLanguages.includes(language)) {
+    showAlert(
+      null,
+      'App need to restart in order to use this language',
+      () => [I18nManager.forceRTL(true), ReactNativeRestart.Restart()],
+      'Restart',
+      null,
+      'Cancel',
+    );
+  }
+
+  // check if is rtl and user choose ltr language
+  if (I18nManager.isRTL && !rtlLanguages.includes(language)) {
+    showAlert(
+      null,
+      'App need to restart in order to use this language',
+      () => [I18nManager.forceRTL(false), ReactNativeRestart.Restart()],
+      'Restart',
```

Comment:
These will need translation.

Reply:
Added for english and arabic needed to be updated for the other languages too as most of them missing alot of translations